Source/ASNetworkImageNode.mm

```diff
@@ -334,8 +335,9 @@ - (void)displayWillStartAsynchronously:(BOOL)asynchronously
   if (asynchronously == NO && _cacheFlags.cacheSupportsSynchronousFetch) {
     ASDN::MutexLocker l(__instanceLock__);
 
-    if (_imageLoaded == NO && _URL && _downloadIdentifier == nil) {
-      UIImage *result = [[_cache synchronouslyFetchedCachedImageWithURL:_URL] asdk_image];
+    NSURL *url = _URL;
+    if (_imageLoaded == NO && url && _downloadIdentifier == nil) {
+      UIImage *result = [[_cache synchronouslyFetchedCachedImageWithURL:url] asdk_image];
```

Comment:
We read `_URL` into a local variable so that it can be preserved after we unlock to call out to the delegate.